kubernetes/cache/kube_cache.go

```diff
@@ -1590,7 +1625,8 @@ func (c *kubeCache) GetK8sGateway(namespace, name string) (*gatewayapi_v1.Gatewa
 	}
 
 	retG := g.DeepCopy()
-	retG.Kind = kubernetes.K8sGatewayType
+	retG.Kind = kubernetes.K8sActualGatewayType
```

Comment:
Should this be changed from `kubernetes.K8sGatewayType`? I thought there were conflicts with using the actual kind? Why did all the gateway api objects have their kinds changed?

Reply:
yes the initial idea of having `kubernetes.K8sGatewayType` was to skip conflicts with "GatewayType", shouldn't the "APIVersion" resolve the cache conflicts?
If it does not, then K8s objects should not be the actual type.

Reply:
As I said previously, OSSMC uses the `Kind` value to create the link to the specific `Gateway API` object. If this is not the actual type, the link created is wrong.

Reply:
In that case, I think #7457 must be fixed previously to avoid such conflicts. IMO, `Kind` and `apiVersion` should be the actual ones.